fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -115,312 +147,241 @@ def method(self):
 
     def tag_leaks(self, tag="leak"):
-        """Tag leaks"""
-        self.leaks_view().tag_samples([tag])
+        """Tags all potential leaks in :meth:`leaks_view` with the given tag.
 
-    def _id2splitConstructor(self):
+        Args:
+            tag ("leak"): the tag string to apply
+        """
+        self.leaks_view().tag_samples(tag)
 
-        # do this once at the beggining of the run
-        # has O(n) memory cost but memory is cheap compared to
-        # doing a couple of `in` operations per sample every run
-        # I want that sweet sweet O(1) lookup
+    def _initialize(self):
         id2split = {}
+
+        split_ids = {}
         for split_name, split_view in self.split_views.items():
-            sample_ids = split_view.values("id")
+            sample_ids = set(split_view.values("id"))
+            split_ids[split_name] = sample_ids
             id2split.update({sid: split_name for sid in sample_ids})
 
-        return id2split
-
-
-def _to_views(samples, split_views=None, split_field=None, split_tags=None):
-    """Helper function so that we can always work with views. I.e. creates
-    the splits denoted by one of the keyword arguments as views.
-
-    Args:
-        - samples: the samples in the datset
-        One and only one of the following:
-        - split_views (None): a dict of the splits as views. If this value is passed
-        it is returned as is
-        - split_field (None): a string corresponding to a field that has a unique
-        string value for each split e.g. samples that are in the 'train' split
-        will have a value of 'train' in this split.
-        - split_tags (None): a list of strings corresponding to the tags of the
-        different splits.
-    """
-
-    arithmetic_true = lambda x: int(x is not None)
-    num_given = (
-        arithmetic_true(split_views)
-        + arithmetic_true(split_field)
-        + arithmetic_true(split_tags)
-    )
+        # Check for overlapping splits
+        split_names = list(split_ids.keys())
+        for idx, split1 in enumerate(split_names):
+            for split2 in split_names[idx + 1 :]:
+                overlap = split_ids[split1] & split_ids[split2]
+                if overlap:
+                    logger.warning(
+                        "The '%s' and '%s' splits contain %d overlapping samples."
+                        "Use dataset.match_tags('%s').match_tags('%s') to "
+                        "identify them",
+                        split1,
+                        split2,
+                        len(overlap),
+                        split1,
+                        split2,
+                    )
 
-    if num_given == 0:
-        raise ValueError(f"One of the split arguments must be given.")
-    if num_given > 1:
-        raise ValueError(f"Only one of the split arguments must be given.")
+        # Check for samples not in index
+        index_ids = self._similarity_index.sample_ids
```

Comment:
The `SimilarityIndex` interface provides a `sample_ids` property:
https://github.com/voxel51/fiftyone-brain/blob/7b0259ce1bc1070050de998679ded723938bbf4f/fiftyone/brain/similarity.py#L339-L342

which is only implemented by backends for which getting the full set of IDs in the index is something that can be done in a reasonable amount of time. So, this implementation will perform the ID verification that you wanted **IF** the similarity backend supports it.

Now, in practice, the only backend that currently supports `SimiliarityIndex.sample_ids` is the sklearn backend, which not coincidentally is the only one that implementation `DuplicatesMixin`, so in practice this ID verification will now always happen.